api_test/CMakeLists.txt

```diff
@@ -5,7 +5,7 @@ add_executable(api_test
   main.c
 )
 if(CMARK_SHARED)
-  target_link_libraries(api_test libcmark)
+  target_link_libraries(api_test cmark)
 else()
-  target_link_libraries(api_test libcmark_static)
+  target_link_libraries(api_test cmark_static)
```

Comment:
Why do the library names change?  Is that essential?  What effect does it have?

Reply:
The name of the target in `CMakeLists.txt` dictates the name of the generated target in the `cmark-config.cmake`. When searching with `find_package(cmark)` in CMake the expectation is a target like `cmark::cmark`. The `cmark::` part is an indication of an imported target (not compiled by your current project, but imported from 'outside'